src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngSymmetricAlgorithmCore.cs

```diff
@@ -235,6 +274,27 @@ private void ValidateFeedbackSize(CipherMode mode, int feedbackSizeInBits)
             }
         }
 
+        private byte[] CopyAndValidateKey(byte[] rgbKey)
+        {
+            byte[] key = rgbKey.CloneByteArray();
```

Comment:
Seems like this could take a ReadOnlySpan and then build the byte[] by rgbKey.ToArray().  And then the ReadOnlySpan-ness could propagate up to the callers.